o11y/tracer/datadog/datadog.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package datadog
+
+import (
+	"strconv"
+
+	"github.com/DataDog/chaos-controller/o11y/tracer/types"
+	"go.opentelemetry.io/otel/trace"
+
+	ddotel "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/opentelemetry"
+	ddtracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Sink describes a datadog tracer sink
+type Sink struct {
+	provider *ddotel.TracerProvider
+}
+
+// New initiated datadog tracer sink
+func New() Sink {
+	provider := ddotel.NewTracerProvider(
+		ddtracer.WithProfilerCodeHotspots(true),
+		ddtracer.WithLogStartup(false),
+	)
+
+	return Sink{provider: provider}
+}
+
+// GetLoggableTraceContext returns a Datadog-friendly trace context for logs
+// It allows to connect logs with corresponding traces and spans
+func (Sink) GetLoggableTraceContext(span trace.Span) []interface{} {
+	stringLogContext := []string{
+		"dd.trace_id", convertTraceID(span.SpanContext().TraceID().String()),
+		"dd.span_id", convertTraceID(span.SpanContext().SpanID().String()),
+	}
+
+	interfaceLogContext := make([]interface{}, len(stringLogContext))
+	for i, v := range stringLogContext {
+		interfaceLogContext[i] = v
+	}
```

Comment:
Just so you know the alternative exists, it's definitely great like you did:
```suggestion
	interfaceLogContext := make([]interface{}, 0, len(stringLogContext))
	for i, v := range stringLogContext {
		interfaceLogContext = append(interfaceLogContext, v)
	}
```